src/platform/Tizen/BLEManagerImpl.cpp

```diff
@@ -65,23 +98,32 @@ static constexpr unsigned kNewConnectionScanTimeoutMs = 10000;
 /* Tizen Default Connect Timeout */
 constexpr System::Clock::Timeout kConnectTimeoutMs = System::Clock::Seconds16(10);
 
+const char * BLEManagerImpl::ConvertErrToStr(int error)
+{
+    for (auto iter : ErrToStr)
```

Comment:
Instead of lookup table it's better to write it as `switch (error) { ... }` statement. Compiler will optimize such syntax to lookup table (if IDs are consecutive), or binary search (which is more efficient than linear search). Another advantage of using `switch` is static check for missing enum values (if `default` is not included). The type of Tizen BT error is `bt_error_e` which is enum.

Reply:
Why switch will increase code repetition? With switch case in the `ConvertErrToStr()` function the `struct ErrToStrTable`  and `static ErrToStrTable ErrToStr[]` won't be required any more, so there will be no duplication of code.